src_c/math.c

```diff
@@ -1304,9 +1318,12 @@ vector_normalize(pgVector *self, PyObject *_null)
     }
     memcpy(ret->coords, self->coords, sizeof(ret->coords[0]) * ret->dim);
 
-    if (!vector_normalize_ip(ret, NULL)) {
+    PyObject *tmp = vector_normalize_ip(ret, NULL);
+    if (!tmp) {
+        Py_DECREF(ret);
         return NULL;
     }
+    Py_DECREF(tmp);
```

Comment:
That's crazy. You have to do it because vector_normalize_ip returns and increments `None`. So you need to decrement it.

Seems like a good opportunity for a helper function to take over the logic of the _ip and non _ip version, but that's out of scope for this PR ofc.